src/esp/assets/ResourceManager.h

```diff
@@ -1216,7 +1227,25 @@ class ResourceManager {
    */
   std::map<std::string, AbstractPrimitiveAttributes::ptr>
       primitiveAssetsTemplateLibrary_;
-
+  /**
+   * @brief Constant Map holding names of all Magnum 3D primitive classes
+   * supported, keyed by @ref PrimObjTypes enum entry.  Note final entry is not
+   * a valid primitive.
+   */
+  const std::map<PrimObjTypes, const char*> PrimitiveNames3DMap = {
+      {PrimObjTypes::CAPSULE_SOLID, "capsule3DSolid"},
+      {PrimObjTypes::CAPSULE_WF, "capsule3DWireframe"},
+      {PrimObjTypes::CONE_SOLID, "coneSolid"},
+      {PrimObjTypes::CONE_WF, "coneWireframe"},
+      {PrimObjTypes::CUBE_SOLID, "cubeSolid"},
+      {PrimObjTypes::CUBE_WF, "cubeWireframe"},
+      {PrimObjTypes::CYLINDER_SOLID, "cylinderSolid"},
+      {PrimObjTypes::CYLINDER_WF, "cylinderWireframe"},
+      {PrimObjTypes::ICOSPHERE_SOLID, "icosphereSolid"},
+      {PrimObjTypes::ICOSPHERE_WF, "icosphereWireframe"},
+      {PrimObjTypes::UVSPHERE_SOLID, "uvSphereSolid"},
+      {PrimObjTypes::UVSPHERE_WF, "uvSphereWireframe"},
+      {PrimObjTypes::END_PRIM_OBJ_TYPES, "NONE DEFINED"}};
```

Comment:
Since this is `const` already, I would make it `static` and put the definition into the cpp file so it doesn't pollute the header too much.